js/ui/viewSelector.js

```diff
@@ -876,20 +828,57 @@ const ViewSelector = new Lang.Class({
     },
 
     _showPage: function(page, doFadeAnimation) {
-        if (!Main.overview.visible)
-            return;
-
         if (page == this._activePage)
             return;
 
         let oldPage = this._activePage;
         this._activePage = page;
         this._pageChanged();
 
-        if (oldPage)
-            this._animateOut(oldPage, doFadeAnimation)
-        else
-            this._animateIn(null, doFadeAnimation);
+        if (oldPage && doFadeAnimation) {
+            // When fading to the apps page, tween the opacity of the
+            // clone instead, and set the apps page to full solid immediately
+            if (page == this._appsPage) {
+                page.opacity = 255;
+                this._overviewViewsClone.opacity = AppDisplay.EOS_INACTIVE_GRID_OPACITY;
+                Tweener.addTween(this._overviewViewsClone,
+                                 { opacity: AppDisplay.EOS_ACTIVE_GRID_OPACITY,
+                                   saturation: AppDisplay.EOS_ACTIVE_GRID_SATURATION,
+                                   time: OverviewControls.SIDE_CONTROLS_ANIMATION_TIME,
+                                   transition: 'easeOutQuad',
+                                   onComplete: function() {
+                                       this._overviewViewsClone.opacity = AppDisplay.EOS_INACTIVE_GRID_OPACITY;
+                                       this._overviewViewsClone.saturation = AppDisplay.EOS_INACTIVE_GRID_SATURATION;
+                                   },
+                                   onCompleteScope: this });
+            }
+
+            // When fading from the apps page, tween the opacity of the
+            // clone instead. The code in this._fadePageIn() will hide
+            // the actual page immediately
+            if (oldPage == this._appsPage) {
+                this._overviewViewsClone.opacity = AppDisplay.EOS_ACTIVE_GRID_OPACITY;
+                this._overviewViewsClone.saturation = AppDisplay.EOS_ACTIVE_GRID_SATURATION;
+                Tweener.addTween(this._overviewViewsClone,
+                                 { opacity: AppDisplay.EOS_INACTIVE_GRID_OPACITY,
+                                   saturation: AppDisplay.EOS_INACTIVE_GRID_SATURATION,
+                                   time: OverviewControls.SIDE_CONTROLS_ANIMATION_TIME,
+                                   transition: 'easeOutQuad' });
+                this._fadePageIn(oldPage, doFadeAnimation);
+            } else {
+                Tweener.addTween(oldPage,
+                                 { opacity: 0,
+                                   time: OverviewControls.SIDE_CONTROLS_ANIMATION_TIME,
+                                   transition: 'easeOutQuad',
+                                   onComplete: Lang.bind(this,
```

Comment:
Use `onCompleteScope` here instead of `Lang.bind()` for consistency with the surrounding code.

Reply:
Again, took from the original code, but will change it. onCompleteScope seems clearer to me than Lang.bind()